### PR TITLE
Feature/cover hmi notifications with unit tests

### PR DESCRIPTION
**Pull request by okozlovlux**
Done:
 - cover files \src\components\application_manager\src\commands\hmi\on_*.cc with unit tests

@LevchenkoS @AByzhynar @Kozoriz @vlantonov @anosach-luxoft @wolfylambova22 please review

### CODE REVIEW

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 1]
```diff
@@ -0,0 +1,2125 @@
...
+    ++id_iterator;
+  }
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnAppUnregisteredNotificationEventDispatcher) {
+  int32_t event_id = hmi_apis::FunctionID::INVALID_ENUM;
+  MessageSharedPtr message = CreateMessage();
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnAppUnregisteredNotification>(message);
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux, Does not would be more readable setup `am::commands` namespace in current file?

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 2]
```diff
@@ -0,0 +1,2125 @@
...
+  EXPECT_CALL(policy_interface_, OnSystemError(_)).WillOnce(GetArg(&code));
+  command->Run();
+  EXPECT_CALL(app_mngr_, GetPolicyHandler()).Times(0);
+  EXPECT_CALL(policy_interface_, OnAppPermissionConsent(_, _)).Times(0);
+  EXPECT_EQ(hmi_apis::Common_SystemError::SYNC_REBOOTED, code);
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnSystemInfoChangedNotificationCheckLanguage) {
+  const uint32_t lang_code = 5u;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux, please, check naming.

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+              commands::OnPolicyUpdate,
+              commands::OnDeviceStateChangedNotification,
+              commands::hmi::OnTTSResetTimeoutNotification,
+              commands::hmi::DialNumberResponse,
+              commands::hmi::DialNumberRequest,
+              commands::OnEventChangedNotification,
+              commands::OnNaviWayPointChangeNotification,
+              commands::OnUILanguageChangeNotification,
+              commands::OnVRLanguageChangeNotification,
+              commands::OnTTSLanguageChangeNotification> HMICommandsListFour;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @okozlovlux HMICommandsListFourth seems better than HMICommandsListFour

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+              commands::OnVIHeadLampStatusNotification,
+              commands::OnVIEngineTorqueNotification,
+              commands::OnVIAccPedalPositionNotification,
+              commands::OnVISteeringWheelAngleNotification,
+              commands::OnVIMyKeyNotification,
+              commands::AllowAllAppsRequest,
+              commands::AllowAllAppsResponse,
+              commands::AllowAppRequest,
+              commands::AllowAppResponse,
+              commands::UISetIconRequest> HMICommandsListFive;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @okozlovlux HMICommandsListFifth seems better than HMICommandsListFive

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 5]
```diff
@@ -0,0 +1,2125 @@
...
+  command->Run();
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnExitApplicationNotificationManageMobileCommand) {
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::strings::app_id] = kAppId_;
+  am::smart_objects::SmartObjectSPtr notification =
+      utils::MakeShared<am::smart_objects::SmartObject>();
+  (*notification)[am::strings::params][am::strings::function_id] =
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux, cast from `enum` to `int32_t` is redundant.

Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @LevchenkoS  @okozlovlux It is NOt redundant. You have to use static_cast to make conversion explicit.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 6]
```diff
@@ -0,0 +1,2125 @@
...
+    ++it;
+    commands_list_manage_mobile_.push_back(
+        CreateCommand<am::commands::OnVIVehicleDataNotification>((*it)));
+    ++it;
+    commands_list_manage_mobile_.push_back(
+        CreateCommand<am::commands::OnVIVinNotification>((*it)));
+    ++it;
+    commands_list_manage_mobile_.push_back(
+        CreateCommand<am::commands::OnVIWiperStatusNotification>((*it)));
+    ++it;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux This operation is not needed.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 7]
```diff
@@ -0,0 +1,2125 @@
...
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnStatusUpdateNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnVideoDataStreamingNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnRecordStartdNotification>((*it)));
+    ++it;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux This operation is not needed.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 8]
```diff
@@ -0,0 +1,2125 @@
...
+    }
+  }
+
+  void FillCommandsListForManageMobile() {
+    std::vector<MessageSharedPtr>::iterator it =
+        messages_list_for_commands_manage_mobile_.begin();
+
+    commands_list_manage_mobile_.push_back(
+        CreateCommand<am::commands::OnVIAccPedalPositionNotification>((*it)));
+    ++it;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux These three lines are repetitive with only one argument changing. Perhaps a macro or template will be useful here.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 9]
```diff
@@ -0,0 +1,2125 @@
...
+    }
+  }
+
+  void FillCommandsListToHmi() {
+    std::vector<MessageSharedPtr>::iterator it =
+        messages_list_for_commands_to_hmi_.begin();
+
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnAppPermissionChangedNotification>((*it)));
+    ++it;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux These three lines are repetitive with only one argument changing. Perhaps a macro or template will be useful here.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 10]
```diff
@@ -0,0 +1,2125 @@
...
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::VR_Started);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::VR_Stopped);
+  }
+
+  void FillListCommandsEventDispatcher() {
+    commands_list_event_dispatcher_.push_back(
+        CreateCommand<am::commands::OnAppActivatedNotification>(
+            CreateMessage()));
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux Repetitive code with only one argument changing. Perhaps a macro or template will be useful here.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 11]
```diff
@@ -0,0 +1,2125 @@
...
+
+TEST_F(HMICommandsNotificationsTest,
+       OnAllowSDLFunctionalityNotificationDeviceKeyNotExist) {
+  const std::string device_id = "";
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::hmi_response::allowed] = true;
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnAllowSDLFunctionalityNotification>(message);
+
+  bool value;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux It is good to initialize this variable

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 12]
```diff
@@ -0,0 +1,2125 @@
...
+TEST_F(HMICommandsNotificationsTest,
+       OnAllowSDLFunctionalityNotificationDeviceKeyExist) {
+  const std::string device_id = "device_id";
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::hmi_response::allowed] = true;
+  (*message)[am::strings::msg_params]["device"]["id"] = device_id;
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnAllowSDLFunctionalityNotification>(message);
+
+  bool value;
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux It is good to initialize this variable

**Comment by LevchenkoS** [comment]
@okozlovlux, reviewed.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 13]
```diff
@@ -0,0 +1,1993 @@
...
+  EXPECT_CALL(mock_hmi_capabilities_, set_active_vr_language(_));
+  EXPECT_CALL(app_mngr_, hmi_capabilities())
+      .WillOnce(ReturnRef(mock_hmi_capabilities_));
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(_, Command::CommandOrigin::ORIGIN_SDL));
+  EXPECT_CALL(*app_ptr_, app_id()).WillOnce(Return(kAppId_));
+  EXPECT_CALL(*app_ptr_, language()).WillRepeatedly(ReturnRef(kLang));
+
+  command->Run();
+  EXPECT_CALL
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux No action set for this expect call

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 6 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 7 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 8 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 9 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 10 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 11 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 12 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc (hunk 13 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @vlantonov fixed in d02f33f

**Comment by okozlovlux** [comment]
@Kozoriz @anosach-luxoft @VVeremjova @wolfylambova22 please review

**Comment by wolfylambova22** [comment]
Reviewed.

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc [hunk 14]
```diff
@@ -0,0 +1,1935 @@
...
+  NiceMock<MockApplication>* app_ptr_;
+
+  const uint32_t kAppId_ = 2014u;
+
+  void SetUp() OVERRIDE {
+    message_helper_mock_ =
+        application_manager::MockMessageHelper::message_helper_mock();
+  }
+
+#define MEDIA true
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc]: @okozlovlux maybe move defines out from class declaration?

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by vlantonov** [comment]
Reviewed.

**Comment by AByzhynar** [comment]
@okozlovlux Squash commits please

**Comment by Kozoriz** [comment]
Reviewed

**Comment by okozlovlux** [comment]
@AByzhynar squash commits have been done

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**